### PR TITLE
#3767 onboarding impreovements

### LINKED ISSUE #3767: Onboarding Improvements
### Overview

This is a smaller-scale epic focused on improving the user experience during onboarding and ensuring it is intuitive for new members. This will be done through a combination of code changes and language / content changes on the onboarding, but our primary objective is to design a flow that makes it very obvious to users what they need to do next at each step.

### Stakeholders

Product Stakeholder: @naishahmistry 
Software Stakeholder: @chpy04 @walker-sean 
Reference Users: Will Laroche

### User Story

As a new member in the club, I want to have a comprehensive onboarding page that contains all the information that I need to know what to do next, so that I do not have to reach out to people to complete onboarding steps, and I am never waiting for someone to tell me what to do next.

### Success Metrics

We will know if these changes are effective by looking at attrition rates for onboarding. The ratio of people who finish onboarding to those who make an account on finishline is a direct metric that should be used to track the success.

### Rollout Plan

This epic combines simple code fixes/language changes and content updates with new pages/workflow steps. The simple changes that are low effort, high value should be prioritized and go live before onboarding starts for Spring 2026, while the larger additions, such as new pages and workflow steps, should be designed and implemented during the Spring 2026 semester and rolled out before Fall 2026 onboarding

### Out of Scope

This epic does not include changing the language in the content of onboarding (other than "application" if we decide to change that).

### Background / Context

We already have an onboarding process, but we saw a lot of people get confused by its purpose and either not use it or get stuck without reaching out for help and leave the club entirely. We want it to be clearer to people what they need to do and accelerate the process of getting people doing actual work in the club.

### Acceptance Criteria & Mock-ups

- The checkboxes are very lagging right now because they don't change until the backend request goes through. Checking a box should be instant (as soon as the box is clicked, it changes right away instead of waiting for the request to go through)
- The top level checkboxes are not meant to be clicked, so they should be visually distinct from the other check boxes and should not look clickable. Clicking them should not do anything whether they are checked or unchecked (right now checking them always just gives an error saying there are unchecked items and unchecking them unchecks all of the child items which is just annoying and would be better to just fully disable the parent check boxes)
- Right now, checklist content consists of a list of checkbox items on the left and then a description on the right. Instead of this, we want the description to be mixed in with the checkboxes in a single combined list, so that we can utilize the full width of the component for both the descriptions and the tasks. In the admin tools under a specific task, the add subtask button should instead say add subtask or information, and clicking it should first prompt the user whether they want to add a checkbox item or a information block. The information block should use a similar form to what the current discription uses, including the markdown viewer. Discriptions and sub tasks in the onboarding page should be displayed as a vertical stack with all items taking up the full width of the task component.
- In the admin tools, the order of tasks and the order of subtasks / info blocks within a task should be adjustable by dragging and dropping with the dots on the left side, and should change the database so that the order is reflected on the frontend
- All links in the onboarding page should open a new tab always, including both external and finishline links

### Tickets

 - [ ] #
 - [ ] #

### PR DESCRIPTION
**Pull request by chpy04**
## Changes

- Can now add either info boxes or check boxxes to the checklists so that we don't have check boxes with paragraphs of text next to them
- Migrated old description fields to just be sub-tasks of info blocks, changed name to content so content is either the text for the checkbox or the text for the info box
- Stores order of tasks and subtasks in db and allows admins to change order
- optimistically updates cache when users click checkboxes to prevent lag
- Removed apply here then continue section per Wills request
- added image for new member events to put on onboarding page bc I felt like it would be useful
- Defaulted checklists on the onboarding page to be expanded

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #3767

### CODE REVIEW

File: src/backend/src/prisma/migrations/20260106000000_add_checklist_ordering_and_type/migration.sql [hunk 1]
```diff
@@ -0,0 +1,112 @@
+-- CreateEnum
+CREATE TYPE "Checklist_Item_Type" AS ENUM ('TASK', 'INFO');
+
+-- AlterTable
+ALTER TABLE "Checklist" ADD COLUMN "displayOrder" INTEGER,
```
Comment (walker-sean) [inline src/backend/src/prisma/migrations/20260106000000_add_checklist_ordering_and_type/migration.sql]: nit: could we do `displayIndex`, fits more with integer type

File: src/backend/src/services/onboarding.services.ts [hunk 2]
```diff
@@ -14,7 +14,12 @@ export default class OnboardingServices {
   static async getAllChecklists(organization: Organization) {
     const allChecklists = await prisma.checklist.findMany({
       where: { organizationId: organization.organizationId, dateDeleted: null, parentChecklistId: null },
-      include: { subtasks: { where: { dateDeleted: null } }, teamType: true, usersChecked: true }
+      include: {
+        subtasks: { where: { dateDeleted: null }, orderBy: { displayOrder: 'asc' } },
+        teamType: true,
+        usersChecked: true
```
Comment (walker-sean) [inline src/backend/src/services/onboarding.services.ts:20]: is this actually used anywhere?

Comment (walker-sean) [reply src/backend/src/services/onboarding.services.ts:20]: Nvm I forgot we displayed checks by checking against this. Ideally we'd have another endpoint call for this instead but it doesn't really matter

File: src/backend/src/services/onboarding.services.ts [hunk 3]
```diff
@@ -166,11 +176,28 @@ export default class OnboardingServices {
...
 
+    // Calculate next displayOrder
+    const existingChecklists = await prisma.checklist.findMany({
+      where: {
+        organizationId: organization.organizationId,
+        parentChecklistId: parentChecklistId ?? null,
+        dateDeleted: null
+      },
+      orderBy: { displayOrder: 'desc' },
+      take: 1
```
Comment (walker-sean) [inline src/backend/src/services/onboarding.services.ts]: you can use findFirst

File: src/backend/src/services/onboarding.services.ts [hunk 4]
```diff
@@ -166,11 +176,28 @@ export default class OnboardingServices {
       }
     }
 
+    // Calculate next displayOrder
+    const existingChecklists = await prisma.checklist.findMany({
```
Comment (walker-sean) [inline src/backend/src/services/onboarding.services.ts]: This will only be one checklist

File: src/backend/src/services/onboarding.services.ts [hunk 5]
```diff
@@ -166,11 +176,28 @@ export default class OnboardingServices {
...
+        dateDeleted: null
+      },
+      orderBy: { displayOrder: 'desc' },
+      take: 1
+    });
+
+    const nextDisplayOrder =
+      existingChecklists.length > 0 && existingChecklists[0].displayOrder !== null
+        ? existingChecklists[0].displayOrder + 1
+        : 1;
```
Comment (walker-sean) [inline src/backend/src/services/onboarding.services.ts]: Why 1-index?

File: src/backend/src/services/onboarding.services.ts [hunk 6]
```diff
@@ -431,4 +464,88 @@ export default class OnboardingServices {
...
+        dateDeleted: null,
+        parentChecklistId: null
+      }
+    });
+
+    if (tasks.length !== taskIds.length) {
+      throw new HttpException(400, 'One or more task IDs are invalid');
+    }
+
+    await prisma.$transaction(async (tx) => {
```
Comment (walker-sean) [inline src/backend/src/services/onboarding.services.ts:489]: nice

File: src/backend/src/services/organizations.services.ts [hunk 7]
```diff
@@ -353,6 +352,56 @@ export default class OrganizationsService {
...
+    submitter: User,
+    organization: Organization
+  ): Promise<Organization> {
+    if (!(await userHasPermission(submitter.userId, organization.organizationId, isAdmin))) {
+      throw new AccessDeniedAdminOnlyException('update new member image');
+    }
+
+    const newMemberImageData = await uploadFile(newMemberImage);
+
+    if (!newMemberImageData?.name) {
```
Comment (walker-sean) [inline src/backend/src/services/organizations.services.ts:375]: Do we know when this would happen? The upload throws when gdrive has an error, right?

File: src/frontend/src/components/NERMarkdown.tsx [hunk 8]
```diff
@@ -1,12 +1,21 @@
...
-import ReactMarkdown from 'react-markdown';
+import ReactMarkdown, { Components } from 'react-markdown';
 import { Box } from '@mui/material';
 
 interface NERMarkdownProps {
   markdown: string;
 }
 
 const NERMarkdown = ({ markdown }: NERMarkdownProps) => {
+  // make all links open in new tabs
```
Comment (walker-sean) [inline src/frontend/src/components/NERMarkdown.tsx:10]: nice

File: src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx [hunk 9]
```diff
@@ -30,119 +34,132 @@ interface CreateChecklistModalProps {
...
-    .array()
-    .of(
-      yup.object().shape({
-        name: yup.string().required('Subtask Name is Required'),
-        isOptional: yup.boolean().required('Is Optional is Required')
-      })
-    )
-    .required()
+const schema: yup.ObjectSchema<ChecklistFormValues> = yup.object().shape({
+  content: yup.string().required('Task name is required')
```
Comment (walker-sean) [inline src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx]: nit: content

File: src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx [hunk 10]
```diff
@@ -30,119 +34,132 @@ interface CreateChecklistModalProps {
...
+  };
+
+  const addInfoBlock = () => {
+    setItems([...items, { id: `info-${Date.now()}`, type: ChecklistItemType.INFO, content: '' }]);
+  };
+
+  const updateItem = (index: number, updates: Partial<ChecklistItem>) => {
+    const newItems = [...items];
+    newItems[index] = { ...newItems[index], ...updates };
+    setItems(newItems);
```
Comment (walker-sean) [inline src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx]: We should prob use the callback form to avoid stale states from being picked up, but if you haven't noticed any issues with quick changes it should be fine

File: src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/EditInfoBlockModal.tsx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+
+interface InfoBlockFormValues {
+  content: string;
+}
+
+const schema = yup.object().shape({
+  content: yup.string().required('Content is required')
+});
+
+const EditInfoBlockModal: React.FC<EditInfoBlockModalProps> = ({ open, handleClose, parentChecklist, defaultValues }) => {
```
Comment (walker-sean) [inline src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/EditInfoBlockModal.tsx]: Dont wanna block this PR but as a follow up we should make a general info block modal

File: src/frontend/src/pages/HomePage/components/ChecklistSection.tsx [hunk 12]
```diff
@@ -13,16 +15,18 @@ interface ChecklistSectionProps {
...
+  const { data: newMemberImageUrl } = useGetImageUrl(organization?.newMemberImageId ?? null);
 
   if (!organization || isLoading) return <LoadingIndicator />;
   if (isError) return <ErrorPage message={error?.message} />;
 
   return (
     <Box>
       <Grid container>
-        {organization.applicationLink && (
+        {/* {organization.applicationLink && (
```
Comment (walker-sean) [inline src/frontend/src/pages/HomePage/components/ChecklistSection.tsx:29]: We can remove this right

**Review by walker-sean — CHANGES_REQUESTED** [review]

File: src/backend/src/services/onboarding.services.ts (hunk 2 above)
Comment (chpy04) [reply src/backend/src/services/onboarding.services.ts:20]: yeah it could be cleaner but IMO this is fine

File: src/backend/src/services/onboarding.services.ts (hunk 5 above)
Comment (chpy04) [reply src/backend/src/services/onboarding.services.ts]: Because people forget that 0 is falsey all the time and it feels like a safeguard in case some code slips through

File: src/backend/src/services/organizations.services.ts (hunk 7 above)
Comment (chpy04) [reply src/backend/src/services/organizations.services.ts:375]: If someone uploaded a different file type that didn't have name in the metadata this would get hit. Not super necessary but I was just copying from the logo endpoint so I think its fine to leave it but I'll add a comment

File: src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx (hunk 10 above)
Comment (chpy04) [reply src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx]: good comms

File: src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/EditInfoBlockModal.tsx (hunk 11 above)
Comment (chpy04) [reply src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/EditInfoBlockModal.tsx]: oh shoot yeah u right

File: src/frontend/src/pages/HomePage/components/ChecklistSection.tsx (hunk 12 above)
Comment (chpy04) [reply src/frontend/src/pages/HomePage/components/ChecklistSection.tsx:29]: I wanted to keep it and just leave it commented out because it's the only place that we actually use the application link in the frontend and it seems reasonable that we might add it back at some point so I wanted future developers to be able to see the original intent. I'll leave a comment
